File: pset6/sentimental-credit/credit.py
from cs50 import get_string

while True:
    card = get_string("Number: ")
    try:
        int(card)
    except ValueError:
        continue

    break

sum = 0;

#multiply every other number
for i in range(2, len(card) + 1, 2):
    num = int(card[-i]) * 2
    if num >= 10:
        sum += 1
    sum += num % 10

for i in range(1, len(card) + 1, 2):
    sum += int(card[-i])

if sum % 10 != 0:
    exit(f"INVALID")

firstDigit = card[0]
secondDigit = card[1]

if (len(card) == 13 or len(card) == 16) and firstDigit == "4":
    exit("VISA")
elif len(card) == 16 and int(firstDigit + secondDigit) in range(51,56,1):
    exit("MASTERCARD")
elif len(card) == 15 and int(firstDigit + secondDigit) in [34, 37]:
    exit("AMEX")
else:
    exit("INVALID")

File: pset1/credit/credit.c
#include <math.h>
#include <stdio.h>

int main(void)
{
    long cardNumber;
    int cardSum = 0;
    int firstDigit, secondDigit;

    do {
        printf("What's your card number?");
        scanf("%li",&cardNumber);
    } while (cardNumber < 1);

    //Takes the sum of every other digit starting from second-to-last
    for (int i = 1; i <= 16; i += 2) {
        int valueSum = (int)((long)(cardNumber / pow(10,i)) % 10) * 2;
        cardSum += (valueSum / 10) + (valueSum % 10);
    }

    for (int i = 0; i <= 16; i += 2) {
        cardSum += (int)((long)(cardNumber / pow(10,i)) % 10);
    }

    if (cardSum % 10 != 0) {
        printf("INVALID\n");
        return 0;
    }

    int lenght = 0;

    for (long n = cardNumber; n != 0; n /= 10) {
        lenght++;
    }

    //Get the first and second digits;
    firstDigit = (int)(cardNumber / pow(10,lenght - 1)) % 10;
    secondDigit = (int)(cardNumber / pow(10,lenght - 2)) % 10;

    //Visa
    //Lenght 13 or 16 and first digit = 4;
    if ((lenght == 13 || lenght == 16) && firstDigit == 4)
    {
        printf("VISA\n");
    }
    //MasterCard
    //Lenght 16 and starts with 51-55;
    else if (lenght == 16 && firstDigit == 5 && (secondDigit > 0 && secondDigit < 6))
    {
        printf("MASTERCARD\n");
    }
    //American Express
    //Lenght 15 and starts with 34 or 37;
    else if (lenght == 15 && firstDigit == 3 && (secondDigit == 4 || secondDigit == 7))
    {
        printf("AMEX\n");
    }
    else {
        printf("INVALID\n");
    }

    return 0;
}